app/services/conversation_service.rb

```diff
@@ -0,0 +1,48 @@
+class ConversationService
+
+  def initialize(user=nil)
+    @user = user
+  end
+
+  def all_for_user()
+    Conversation.where(author_id: @user.person.id).all
+  end
+
+  def build(subject, text, recipients)
+    person_ids = @user
+      .contacts
+      .mutual
+      .where(:person_id => recipients)
+      .pluck(:person_id)
+
+    opts = {
+      subject: subject,
+      message: { text: text },
+      participant_ids: person_ids
+    }
+
+    @user.build_conversation(opts)
+  end
+
+  def find!(conversation_id)
+    @user.conversations
+      .joins(:conversation_visibilities)
+      .where(conversation_visibilities: {
+        person_id: @user.person_id,
```

Comment:
- Align the elements of a hash literal if they span more than one line.
- Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.